platform/platform.go

```diff
@@ -0,0 +1,24 @@
+package platform
+
+import (
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/cmd"
+	v05 "github.com/buildpacks/lifecycle/platform/v05"
+	v06 "github.com/buildpacks/lifecycle/platform/v06"
+)
+
+type Platform interface {
+	API() string
+	CodeFor(errType cmd.LifecycleExitError) int
+}
+
+func NewPlatform(apiStr string) Platform {
+	var platform Platform
+	platformAPI := api.MustParse(apiStr)
+	if platformAPI.Compare(api.MustParse("0.5")) > 0 { // platform API > 0.5
+		platform = v06.NewPlatform(apiStr)
+	} else {
+		platform = v05.NewPlatform(apiStr)
+	}
+	return platform
```

Comment:
```suggestion
	platformAPI := api.MustParse(apiStr)
	if platformAPI.Compare(api.MustParse("0.6")) < 0 { // platform API < 0.6
		return v05.NewPlatform(apiStr)
	}
	return v06.NewPlatform(apiStr)
```

I would change the order of the condition because if at some point we stop supporting api < 0.6, it'll be a bit easier to change the code.

Reply:
Nice!